### PR TITLE
Controls: Improve accessibility of BooleanControl for screen readers

### LINKED ISSUE #24128: [Bug]: While navigating to “True/False” button in controls pane narrator announcing as true false check box checked which is incorrect.
### Describe the bug

While navigating to “True/False” button in controls pane narrator announcing as true false check box checked which is incorrect.

Actual Result:
While navigating to “True/False” button in controls pane narrator announcing as true false check box checked which is incorrect.

Expected Result:
While navigating to “True/False” button narrator should announce true checked/unchecked or false checked/unchecked.

### To Reproduce

                     
	
	
 1. Open URL: [Common Navigation - Scrolling Panel ⋅ Storybook (windows.net)](https://vivacommonstorybooktest.blob.core.windows.net/$web/index.html?path=/story/common-navigation--scrolling-panel&globals=backgrounds.grid:!false;backgrounds.value:transparent) 
2. Navigate to “App bar” button at right side of the pane invoke it and navigate to “Default” link invoke it. 
3. Now navigate to “Controls” tab item invoke it and navigate to “True/False” button and observe the narrator announcement.

Attachment:

https://github.com/storybookjs/storybook/assets/93735775/a38273ab-2032-48f3-bc50-f0456d95977b

### System

```bash
OS: Windows 11
OS version: 22H2 (OS Build 22621.2134)
Edge Version: 116.0.1938.69 (Official build) (64-bit)
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by danielmarcano**
Closes [#24128](https://github.com/storybookjs/storybook/issues/24128)

## What I did

- Fixed the accessibility of the `BooleanControl` component, based on the W3C Web Accessibility Initiative recipes for the [Switch component](https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/). This way, screen readers read a more accurate label.

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by JReinhold — APPROVED** [review]
this is great, thank you! could you provide a short audio/video clip of how the screen reader narrates it with this fix?

**Comment by danielmarcano** [comment]
> this is great, thank you! could you provide a short audio/video clip of how the screen reader narrates it with this fix?

Hi, @JReinhold, thank you for the review.

Here's a short video showcasing the way this one is read by Voice Over on macOS:

https://github.com/storybookjs/storybook/assets/93147810/b09143b2-d3b4-4604-8628-20e19ff44f1c

In synth, the component is now read as:

- A switch, which indicates the off/on nature to the user
- The value is mentioned correctly
- The name of the property that this switch relates to is now part of the description as well

Let me know if this would be good to go ✨

**Review by cdedreuille — APPROVED** [review]

File: code/ui/blocks/src/controls/Boolean.tsx [hunk 1]
```diff
@@ -112,16 +119,17 @@ export const BooleanControl: FC<BooleanProps> = ({ name, value, onChange, onBlur
   const parsedValue = typeof value === 'string' ? parse(value) : value;
 
   return (
-    <Label htmlFor={controlId} title={parsedValue ? 'Change to false' : 'Change to true'}>
+    <Label htmlFor={controlId} aria-label={name}>
```
Comment (yannbf) [inline code/ui/blocks/src/controls/Boolean.tsx:122]: Does the title need to be removed? It added a tooltip which was useful for any user

Comment (yannbf) [reply code/ui/blocks/src/controls/Boolean.tsx:122]: In fact, we can improve the title further and provide the name of the control which will be changed:
image

And the voiceover (seems to be) even more helpful:
image

While behaving similarly when interacting with the control:
image

Comment (danielmarcano) [reply code/ui/blocks/src/controls/Boolean.tsx:115]: Hi, @yannbf.

I have basically applied the [Switch Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/switch/), as described by the W3C Web Accessibility Initiative, which states that:

- It is critical the label on a switch does not change when its state changes
- The switch has an accessible label provided by one of the following:
  - Visible text content contained within the element with role switch.
  - A visible label referenced by the value of `aria-labelledby` set on the element with role switch.
  - `aria-label` set on the element with role switch.

The idea is letting the semantic elements and corresponding ARIA roles to let screen reader users know how to use the component, and having a stable label for the element that does not change based on the current value of the interactive element.

That is why I have removed the old labels "Change to true" and "Change to false", and just left a stable one, which is the name of the control which will be changed.

That is also the reason why I have removed the `title` attribute, as:
  - It was not necessary for sighted users, as there's enough visual context to understand its behavior
  - It was no longer necessary for screen reader users, as `aria-label` and `role="switch"` are already conveying the information on both its usage, and the control they relate with

There's also a small excerpt on MDN mentioning that "While title can be used to provide a programmatically associated label for an <input> element, this is not good practice. Use a <label> instead." You can read it [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title).

Knowing this, do you consider these changes to be okay?

**Comment by danielmarcano** [comment]
@yannbf, I noticed that after your last two commits, two checks have stopped working. I'm currently unable to identify the cause of these issues, as the errors appear to be somewhat ambiguous. Could you please take a look at them when you have some time?

File: code/ui/blocks/src/controls/Boolean.tsx (hunk 1 above)
Comment (yannbf) [reply code/ui/blocks/src/controls/Boolean.tsx:115]: Hey @danielmarcano thanks a lot for all the clarification. I think it makes sense, let's proceed without the title!

**Review by yannbf — APPROVED** [review]

**Pull request merged**